admin/services/data-access/headteacher-declaration.data.service.js

```diff
@@ -49,4 +49,98 @@ headteacherDeclarationDataService.sqlFindHdfForCheck = async (dfeNumber, checkWi
   return R.head(result)
 }
 
+/**
+ * Find count of pupils blocking hdf submission before check end date
+ * @param dfeNumber
+ * @return {Number}
+ */
+headteacherDeclarationDataService.sqlFindPupilsBlockingHdfBeforeCheckEndDate = async (dfeNumber) => {
+  const sql = `
+    SELECT COUNT(p.id) as pupilsCount
+    FROM [mtc_admin].[pupil] p
+    JOIN [mtc_admin].[school] s ON p.school_id = s.id
+    LEFT JOIN [mtc_admin].[pupilStatus] ps ON (p.pupilStatus_id = ps.id)
+    WHERE s.dfeNumber = @dfeNumber
+    AND ps.code NOT IN ('NOT_TAKING', 'COMPLETED')
+  `
+
+  const params = [
+    { name: 'dfeNumber', type: TYPES.Int, value: dfeNumber }
+  ]
+
+  const result = await sqlService.query(sql, params)
+  return R.path(['pupilsCount'], R.head(result))
+}
+
+/**
+ * Find count of pupils blocking hdf submission before check end date
+ * @param dfeNumber
+ * @return {Number}
+ */
+headteacherDeclarationDataService.sqlFindPupilsBlockingHdfAfterCheckEndDate = async (dfeNumber) => {
+  const sql = `
+    SELECT COUNT(p.id) as pupilsCount
+    FROM [mtc_admin].[pupil] p
+    JOIN [mtc_admin].[school] s ON p.school_id = s.id
+    LEFT JOIN [mtc_admin].[pupilStatus] ps ON (p.pupilStatus_id = ps.id)
```

Comment:
Again, not sure why a left join is used here.  Please explain.

Reply:
@jon-shipley same.